pkg/build/tempdirs.go

```diff
@@ -83,7 +82,26 @@ func (d TempDirs) VerificationResultCachePath(configID, zipName string) string {
 }
 
 func (d TempDirs) cacheDir(kind string, extension ...string) string {
-	return d.tempDirPath(prefix(extension, "cache", kind, d.product.Repository, d.product.Name, d.product.SourceHash)...)
+	sh := d.product.SourceHash
+	// If the sourcehash is a long hex string, we shorten it and prefix with
+	// dirty_ or clean_. This doesn't matter for cache invalidation as the
+	// hash of the build inputs already includes this.
+	matched, err := regexp.Match("[a-f0-9]{40}.*", []byte(sh))
+	if err != nil {
+		// This panic will only happen if the code above is wrong (i.e. the
+		// regexp doesn't compile). So this won't happen in releases since
+		// they will have passed tests that assert this panic can't happen.
+		panic(err)
+	}
+	if matched {
+		sh = sh[:8]
+		if d.product.IsDirty() {
+			sh = fmt.Sprintf("dirty_%s", sh)
+		} else {
+			sh = fmt.Sprintf("clean_%s", sh)
+		}
+	}
```

Comment:
This little tweak to cache paths lets us quickly see from the path whether the build was considered to be dirty or not. It also makes the paths a bit shorter which aids reading the logs.